staging/src/k8s.io/apimachinery/pkg/util/cache/lruexpirecache.go

```diff
@@ -67,36 +69,71 @@ type cacheEntry struct {
 func (c *LRUExpireCache) Add(key interface{}, value interface{}, ttl time.Duration) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	c.cache.Add(key, &cacheEntry{value, c.clock.Now().Add(ttl)})
+
+	if c.evictionList.Len() >= c.maxSize {
+		toEvict := c.evictionList.Back()
+		c.evictionList.Remove(toEvict)
+		delete(c.entries, toEvict.Value.(*cacheEntry).key)
+	}
+
+	entry := &cacheEntry{
+		key:        key,
+		value:      value,
+		expireTime: c.clock.Now().Add(ttl),
+	}
+
+	element := c.evictionList.PushFront(entry)
```

Comment:
does this handle Add being called multiple times for the same item like c.cache.Add did?

Reply:
it doesn't look like it... need a unit test to verify this